python/smashbox/multiprocessing_engine.py

```diff
@@ -194,20 +193,31 @@ def run():
         _smash_.process_name = "supervisor"
 
         _smash_.steps = manager.list([0 for x in range(len(_smash_.workers))])
+        
+        #report persistent information
+        import smashbox.reporter
+        reporter = smashbox.reporter.Reporter(os.path.basename(_smash_.args.test_target),config)
+        reporter.testcase_start()
 
         # first worker => process number == 0
         for i,f_n in enumerate(_smash_.workers):
             f = f_n[0]
             fname = f_n[1]
-            p = Process(target=_smash_.worker_wrap,args=(i,f,fname))
+            if fname is None:
+                fname = f.__name__
+            reporter.shared_results_manager(manager,fname)
+            p = Process(target=_smash_.worker_wrap,args=(i,f,fname,reporter))
             p.start()
             _smash_.all_procs.append(p)
 
         _smash_.supervisor(_smash_.steps)
-
+        
         for p in _smash_.all_procs:
             p.join()
-
+        
+        #finish the reporter
+        reporter.testcase_stop()
+        
```

Comment:
New reporter will be only functioning for the test (testcase_start and testcase_stop), no need for handling of whole the test.